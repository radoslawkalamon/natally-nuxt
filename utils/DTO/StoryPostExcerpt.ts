import PostExcerpt from '@/utils/DTO/PostExcerpt'
import formatReadingTime from '@/utils/formatters/readingTime'

type StoryPostExcerptConstructor = {
  id: string;
  coverImage: string;
  coverImage2x: string;
  datePublished: number;
  isAudiobook: boolean;
  readingTime: number;
  title: string;
  url: string;
}

export default class StoryPostExcerpt extends PostExcerpt {
  readonly coverImage: string;
  readonly coverImage2x: string;
  readonly isAudiobook: boolean;
  readonly readingTime: string;

  constructor ({ id, coverImage, coverImage2x, datePublished, isAudiobook, readingTime, title, url }: StoryPostExcerptConstructor) {
    super({ id, datePublished, title, url })
    this.coverImage = coverImage
    this.coverImage2x = coverImage2x
    this.isAudiobook = isAudiobook
    this.readingTime = formatReadingTime(readingTime)
  }
}
