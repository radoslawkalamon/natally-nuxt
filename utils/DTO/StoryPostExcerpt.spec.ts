import StoryPostExcerpt from '@/utils/DTO/StoryPostExcerpt'

jest.mock('@/utils/formatter.date.published', () => () => 'Mocked date formatter')
jest.mock('@/utils/formatter.time.reading', () => () => 'Mocked time formatter')

describe('Utils / DTO / Poem Post Excerpt', () => {
  let storyPostExcerpt: StoryPostExcerpt

  beforeAll(() => {
    storyPostExcerpt = new StoryPostExcerpt({
      id: '123',
      coverImage: '/cover-image-url.webp',
      coverImage2x: '/cover-image-url-2x.webp',
      datePublished: 1634302144000,
      isAudiobook: true,
      readingTime: 10,
      title: 'Test title',
      url: '/test-url'
    })
  })

  test('shall format data properly', () => {
    expect(storyPostExcerpt).toMatchSnapshot({
      id: '123',
      coverImage: '/cover-image-url.webp',
      coverImage2x: '/cover-image-url-2x.webp',
      datePublished: 'Mocked date formatter',
      isAudiobook: true,
      readingTime: 'Mocked time formatter',
      title: 'Test title',
      url: '/test-url'
    })
  })
})
