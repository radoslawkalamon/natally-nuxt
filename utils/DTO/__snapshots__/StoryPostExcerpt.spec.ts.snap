// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils / DTO / Poem Post Excerpt shall format data properly 1`] = `
Object {
  "coverImage": "/cover-image-url.webp",
  "coverImage2x": "/cover-image-url-2x.webp",
  "datePublished": "Mocked date formatter",
  "id": "123",
  "isAudiobook": true,
  "readingTime": "Mocked time formatter",
  "title": "Test title",
  "url": "/test-url",
}
`;
